### ISSUE TITLE
Normal class with untyped argument(s) in `__init__` is incorrectly treated as generic

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#1859]
## Environment data

-   Language Server version: 2021.9.3-pre.1
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.11 Anaconda

# Sample code

```python
class C:
    def __init__(self, value):
        ...

c: C[str]
```

## Expected behavior

The type `C[str]` is invalid because `C` is not generic.

## Actual behavior

No errors / warnings.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202193-22-september-2021
